rclcpp/test/rclcpp/test_timer.cpp

```diff
@@ -151,3 +152,43 @@ TEST_F(TestTimer, test_run_cancel_timer)
   EXPECT_TRUE(has_timer_run.load());
   EXPECT_TRUE(timer->is_canceled());
 }
+
+TEST_F(TestTimer, test_bad_arguments) {
+  auto callback = []() {};
+  auto node_base = rclcpp::node_interfaces::get_node_base_interface(test_node);
+  auto context = node_base->get_context();
+
+  auto steady_clock = std::make_shared<rclcpp::Clock>(RCL_STEADY_TIME);
+
+  // Negative period
+  EXPECT_THROW(
+    rclcpp::GenericTimer<decltype(callback)>(steady_clock, -1ms, std::move(callback), context),
```

Comment:
@brawner here and elsewhere, no need for `std::move`:
```suggestion
    rclcpp::GenericTimer<decltype(callback)>(steady_clock, -1ms, callback, context),
```

Sorry if it seemed like I was suggesting so.

Reply:
An rvalue reference is required to pass callback and I was using the decltype() to be lazy about the template instantiation. But I'm not sure what the expected behavior is of passing std::move on an empty callback multiple times. Switching template type to actual type, and passing in an empty callback.